src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -157,24 +157,49 @@ private static bool ApplyChangesFromTC35(JObject sarifLog)
             // https://github.com/oasis-tcs/sarif-spec/issues/366
             modifiedLog |= ConvertAllToolComponentArtifactIndicesToArtifactLocations(sarifLog);
 
-            modifiedLog |= RenameDefaultFileEncodingToDefaultEncoding(sarifLog);
+            if (sarifLog["runs"] is JArray runs)
+            {
+                foreach (JObject run in runs)
+                {
+                    modifiedLog |= RenameProperty(run, "defaultFileEncoding", "defaultEncoding");
+
+                    // https://github.com/oasis-tcs/sarif-spec/issues/375
+                    modifiedLog |= HoistIdsFromPhysicalLocationToLocation(run);
+                }
+            }
 
             return modifiedLog;
         }
 
-        private static bool RenameDefaultFileEncodingToDefaultEncoding(JObject sarifLog)
+        private static bool HoistIdsFromPhysicalLocationToLocation(JObject run)
         {
-            bool isModified = false;
+            string[] locationPathsToUpdate =
+            {
+                "results[].locations",
```

Comment:
>"results[].locations", [](start = 16, length = 22)

we have few more location object paths, are they dont care for us? 

eg. all instances of 

..threadFlowLocation.location,
..stackFrame.location

Reply:
No known SARIF producer emits ids for any other locations! That's why i short-handed this. :)

---
In reply to: [276493698](https://github.com/Microsoft/sarif-sdk/pull/1436#discussion_r276493698) [](ancestors = 276493698)